src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -462,6 +505,54 @@ int encrypt(byte[] in, int inOfs, int len, byte[] out, int outOfs) {
         return len;
     }
 
+    int decrypt(ByteBuffer src, ByteBuffer dst) {
+        if (src.remaining() > 0) {
+            byte[] b = new byte[src.remaining()];
+            src.get(b);
+            try {
+                ibuffer.write(b);
+            } catch (IOException e) {
+                throw new ProviderException("Unable to add remaining input to the buffer", e);
+            }
+        }
+        return 0;
+    }
+
+
+    int encrypt(ByteBuffer src, ByteBuffer dst) {
+        int len = src.remaining();
+        if (len == 0) {
+            return 0;
+        }
+
+        ArrayUtil.blockSizeCheck(src.remaining(), blockSize);
+        checkDataLength(processed, src.remaining());
+        processAAD();
+
+        if (len >= AES_BLOCK_SIZE) {
+            ByteBuffer data = src.duplicate();
+            len = gctrPAndC.update(data, dst);
+        } else {
+            if (src.remaining() > 0) {
+                byte[] b = new byte[src.remaining()];
+                src.get(b);
+                try {
+                    ibuffer.write(b);
+                } catch (IOException e) {
+                    throw new ProviderException(
+                        "Unable to add remaining input to the buffer", e);
+                }
+            }
```

Comment:
Existing usage for ibuffer is only for decryption. If you are storing into ibuffer for encryption, the comment about it should be updated. The earlier code in this method should also check ibuffer and if non-empty, apply its content before applying src?

Reply:
I updated the comment.  I had redone this method before your comment.  Take a look at that on the next webrev